fertiscan/db/queries/label/__init__.py

```diff
@@ -84,35 +93,36 @@ def get_label_information(cursor, label_information_id):
     - label_information_id (str): The UUID of the label_information.
 
     Returns:
-    - dict: The label_information
-    """
-    try:
-        query = """
-            SELECT 
-                id,
-                product_name,
-                lot_number, 
-                npk, 
-                registration_number, 
-                n, 
-                p, 
-                k, 
-                guaranteed_title_en,
-                guaranteed_title_fr,
-                title_is_minimal,
-                company_info_id,
-                manufacturer_info_id
-            FROM 
-                label_information
-            WHERE 
-                id = %s
-            """
-        cursor.execute(query, (label_information_id,))
-        return cursor.fetchone()
-    except Exception as e:
-        raise e
-
+    - dict: A dictionary containing the label information.
 
+    Raises:
+    - InspectionRetrievalError: Custom error for handling retrieval issues.
+    """
+    query = """
+        SELECT 
+            id,
+            product_name,
+            lot_number, 
+            npk, 
+            registration_number, 
+            n, 
+            p, 
+            k, 
+            guaranteed_title_en,
+            guaranteed_title_fr,
+            title_is_minimal,
+            company_info_id,
+            manufacturer_info_id
+        FROM 
+            label_information
+        WHERE 
+            id = %s
+        """
+    cursor.execute(query, (label_information_id,))
+    return cursor.fetchone()
+
+
+@handle_query_errors(LabelInformationRetrievalError)
```

Comment:
I dont understand here we use `LabelInformationRetrievalError` but we raise `LabelInformationNotFoundError` (1/2)

Reply:
Yes. It only means that both could be raised by the same function: LabelInformationNotFoundError in the particular case where None is returned and LabelInformationRetrievalError in the other cases.